src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackageArgs.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using NuGet.Common;
+using NuGet.Configuration;
+using System.Threading;
+
+namespace NuGet.CommandLine.XPlat
+{
+    public class ListPackageArgs
+    {
+        public ILogger Logger { get; }
+        public string Path { get; set; }
+        public List<PackageSource> PackageSources { get; set; }
+        public IList<string> Frameworks { get; set; }
+        public bool IncludeOutdated { get; set; }
+        public bool IncludeDeprecated { get; set; }
+        public bool IncludeTransitive { get; set; }
+        public bool Prerelease { get; set; }
+        public bool HighestPatch { get; set; }
+        public bool HighestMinor { get; set; }
```

Comment:
you don't want a setter on most of these properties. they shouldn't be changed once set from the constructor .

Reply:
I will add this in the push after the last one